### PR TITLE
Roi heading fixes

### PR DESCRIPTION
**Pull request by ndepal**
Adresses https://github.com/PX4/Firmware/issues/8541

Tested in SITL.

### CODE REVIEW

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -450,7 +450,13 @@ MissionBlock::issue_command(const mission_item_s &item)
 		vcmd.param4 = item.params[3];
 		vcmd.param5 = item.params[4];
 		vcmd.param6 = item.params[5];
-		vcmd.param7 = item.params[6];
+
+		if (item.nav_cmd == NAV_CMD_DO_SET_ROI && item.altitude_is_relative) {
```
Comment (ndepal) [inline src/modules/navigator/mission_block.cpp]: Not sure if this is the best way to deal with the relative alt problem.

Comment (LorenzMeier) [reply src/modules/navigator/mission_block.cpp]: Looks sensible.

Comment (dagar) [reply src/modules/navigator/mission_block.cpp]: It's fine for now. Eventually we need to properly use the frame.

**Comment by dagar** [comment]
Looks good, but I think `MNT_YAW_CTL` should actually be a Navigator mission parameter.
 1. The behaviour only applies to ROI evaluated via mission.
 2. The parameter is culled from the system if the module isn't included. That means if vmount isn't included in a build (already not in some) at a minimum every boot will have nasty navigator errors, plus the possibility of unintended behaviour.

**Comment by LorenzMeier** [comment]
Rebased here: https://github.com/PX4/Firmware/pull/8620

**Pull request closed without merging**